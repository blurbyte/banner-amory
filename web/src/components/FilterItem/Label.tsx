import styled from 'styled-components';
import { transparentize } from 'polished';

import { Colors } from '@styles/colors';

type LabelProps = {
  checked?: boolean;
};

const Label = styled.label<LabelProps>`
  position: relative;
  display: inline-flex;
  justify-content: center;
  height: 2rem;
  line-height: 2rem;
  min-width: 2rem;
  background: ${props => (props.checked ? Colors.red : transparentize(0.8, Colors.blue))};
  color: ${Colors.white};
  font-weight: 400;
  cursor: pointer;
  padding: 0 0.6rem;
  margin-right: 1px;
  text-transform: capitalize;
`;

export default Label;
