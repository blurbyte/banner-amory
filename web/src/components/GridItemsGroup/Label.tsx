import styled from 'styled-components';

import Headline from '../Headline';

const Paragraph = Headline.withComponent('p');

const Label = styled(Paragraph)`
  text-transform: uppercase;
  margin: 0;
`;

export default Label;
