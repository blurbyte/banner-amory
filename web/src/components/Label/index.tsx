import styled from 'styled-components';
import { transparentize } from 'polished';

import { Colors } from '@styles/colors';

const Label = styled.label`
  font-size: 1.2rem;
  color: ${transparentize(0.5, Colors.blue)};
  text-transform: uppercase;
  letter-spacing: 0.5px;
`;

export default Label;
