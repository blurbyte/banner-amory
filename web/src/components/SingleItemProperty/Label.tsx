import styled from 'styled-components';
import { transparentize } from 'polished';

import { Colors } from '../../styles/colors';

type LabelProps = {
  statLine?: boolean;
};

const Label = styled.dt<LabelProps>`
  font-size: 1.2rem;
  color: ${transparentize(0.5, Colors.blue)};
  text-transform: uppercase;
  letter-spacing: 0.5px;
  text-align: ${props => props.statLine && 'right'};
  line-height: ${props => (props.statLine ? 1.2 : 'inherit')};
`;

export default Label;
