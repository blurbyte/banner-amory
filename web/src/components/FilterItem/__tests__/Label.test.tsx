import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Label from '../Label';

test('renders correctly', () => {
  const component = renderer.create(<Label />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders correctly when checked', () => {
  const component = renderer.create(<Label checked />);
  expect(component.toJSON()).toMatchSnapshot();
});
