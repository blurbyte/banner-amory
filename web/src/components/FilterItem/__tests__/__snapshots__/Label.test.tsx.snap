// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  line-height: 2rem;
  min-width: 2rem;
  background: rgba(61,38,218,0.2);
  color: #ffffff;
  font-weight: 400;
  cursor: pointer;
  padding: 0 0.6rem;
  margin-right: 1px;
  text-transform: capitalize;
}

<label
  className="c0"
/>
`;

exports[`renders correctly when checked 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 2rem;
  line-height: 2rem;
  min-width: 2rem;
  background: #d92f29;
  color: #ffffff;
  font-weight: 400;
  cursor: pointer;
  padding: 0 0.6rem;
  margin-right: 1px;
  text-transform: capitalize;
}

<label
  checked={true}
  className="c0"
/>
`;
